### PR TITLE
[llvm][OpenMP][NFC] Cleanup AtomicInfo

### PR DESCRIPTION
**Pull request by NimishMishra**
This PR refactors functionality from llvm/include/llvm/Frontend/Atomic/Atomic.h into llvm/lib/llvm/Frontend/Atomic/Atomic.cpp.

### CODE REVIEW

File: llvm/include/llvm/Frontend/Atomic/Atomic.h [hunk 1]
```diff
@@ -62,90 +63,28 @@ template <typename IRBuilderTy> struct AtomicInfo {
...
-    llvm::AttrBuilder fnAttrBuilder(ctx);
-    fnAttrBuilder.addAttribute(llvm::Attribute::NoUnwind);
-    fnAttrBuilder.addAttribute(llvm::Attribute::WillReturn);
-    llvm::AttributeList fnAttrs = llvm::AttributeList::get(
-        ctx, llvm::AttributeList::FunctionIndex, fnAttrBuilder);
-    FunctionCallee LibcallFn = M->getOrInsertFunction(fnName, FnType, fnAttrs);
-    CallInst *Call = Builder->CreateCall(LibcallFn, Args);
-    return Call;
-  }
+  CallInst *EmitAtomicLibcall(IRBuilderBase *Builder, StringRef fnName,
```
Comment (nikic) [inline llvm/include/llvm/Frontend/Atomic/Atomic.h:71]: Does this need the Builder argument if it's already part of the class?

File: llvm/include/llvm/Frontend/Atomic/Atomic.h [hunk 2]
```diff
@@ -12,44 +12,45 @@
...
 #include "llvm/IR/DataLayout.h"
+#include "llvm/IR/IRBuilder.h"
 #include "llvm/IR/Instructions.h"
 #include "llvm/IR/Intrinsics.h"
 #include "llvm/IR/Module.h"
 #include "llvm/IR/Operator.h"
 #include "llvm/IR/RuntimeLibcalls.h"
 
 namespace llvm {
+class AtomicInfo;
```
Comment (nikic) [inline llvm/include/llvm/Frontend/Atomic/Atomic.h:23]: Redundant forward declare?

File: llvm/include/llvm/Frontend/Atomic/Atomic.h [hunk 3]
```diff
@@ -12,44 +12,45 @@
 
 #include "llvm/ADT/DenseMap.h"
 #include "llvm/IR/DataLayout.h"
+#include "llvm/IR/IRBuilder.h"
 #include "llvm/IR/Instructions.h"
 #include "llvm/IR/Intrinsics.h"
 #include "llvm/IR/Module.h"
 #include "llvm/IR/Operator.h"
 #include "llvm/IR/RuntimeLibcalls.h"
```
Comment (nikic) [inline llvm/include/llvm/Frontend/Atomic/Atomic.h:20]: Maybe double check that you still need these includes in the header? E.g. I don't think you need RuntimeLibcalls.h?

**Review by nikic — APPROVED** [review]
LGTM, thanks for following up on this!

**Comment by kparzysz** [comment]
Please fix the comments at the top of both files.  Both say "Atomic.h", and the first line is broken into two.

I think that too many includes are now removed.  For example, the header file uses `IntegerType::get`, but there is no `#include "llvm/IR/DerivedTypes.h"`, or `#include <utility>` for `std::pair`.  I believe the preference is to IWYU, but I'll defer to @nikic to decide.

**Comment by NimishMishra** [comment]
> Please fix the comments at the top of both files. Both say "Atomic.h", and the first line is broken into two.
> 
> I think that too many includes are now removed. For example, the header file uses `IntegerType::get`, but there is no `#include "llvm/IR/DerivedTypes.h"`, or `#include <utility>` for `std::pair`. I believe the preference is to IWYU, but I'll defer to @nikic to decide.

Thanks. I have made the changes.

**Pull request merged**